src/providers/postgres/qgsconnectionconfigurator.h

```diff
@@ -0,0 +1,190 @@
+/***************************************************************************
+  qgspostgresconnectionconfigurator.h  -  connection configurator class for
+  PostgeSQL based connectors.
+                             -------------------
+    begin                : 2023/06/06
+    copyright            : (C) 2023 by Alexey Karandashev
+    email                : reflectored at pm dot me
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSCONNECTIONCONFIGURATOR_H
+#define QGSCONNECTIONCONFIGURATOR_H
+
+#include <QString>
+#include <QtCore/qstring.h>
+
+#include "qgsdatasourceuri.h"
+#include "qgssettings.h"
+
+template <typename T>
+class QgsConnectionConfigurator
+{
+    static const QString connections_prefix;
+    static const QString connection_name_prefix;
+
+  public:
+    static QStringList connectionList()
+    {
+      QgsSettings settings;
+      settings.beginGroup( connections_prefix );
```

Comment:
The settings hierarchy has been recently refactored, and it is exposed to users in the advanced configuration options, you might want to check with @3nids for the best approach here.

Reply:
yes, you have the documentation here: https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html#settings

you can take example on the OWS connections (WMS, WFS, …) which is mostly identical: https://github.com/qgis/QGIS/blob/b4613b98397a08a054ee712292c15b84e08621b1/src/core/qgsowsconnection.h#L96